src/adapter/src/coord/message_handler.rs

```diff
@@ -67,7 +67,44 @@ impl<S: Append + 'static> Coordinator<S> {
             Message::LinearizeReads(pending_read_txns) => {
                 self.message_linearize_reads(pending_read_txns).await;
             }
+            Message::StorageUsage => {
+                self.storage_usage_update().await;
+            }
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn storage_usage_update(&mut self) {
+        let object_id = None;
+        let shard_sizes = self.storageusageclient.shard_sizes().await;
+
+        let mut unk_storage = 0u64;
+        let mut known_storage = 0u64;
+        for (key, val) in shard_sizes.iter() {
```

Comment:
`iter()` shouldn't be needed here since `shard_sizes` is a HashMap which implements `IntoIterator`